README.md

```diff
@@ -28,13 +27,298 @@ pip install -U pyinstaller-hooks-contrib
+then we also have a CLI script to launch it.
+Run ``python scripts/cloud-test.py --help`` which should walk you through it.
+You'll still have to enter all the details again but, thanks to terminal history,
+rerunning a configuration is just a case of pressing up then enter.
+
+
+### Run Linter
+
+We use `flake8` to enforce code-style.
+`pip install flake8` if you haven't already then run it with the following.
+Note that this assumes that you did create a new branch in the [setup step](#setup).
+
+```
+git diff -U0 master | flake8 --diff -
+```
+
+No news is good news.
+If it complains about your changes then do what it asks then run it again.
+If you don't understand the errors it come up with them lookup the error code
+in each line (a capital letter followed by a number e.g. `W391`).
+
+If it complains about code which you haven't written,
+or if you didn't create a new branch at the start then, using `git log`,
+find the commit ID of the newest commit which you didn't write, copy it
+and replace `master` in the above command with that ID.
+
+```
+git diff -U0 a5d3841c282fa23fd68c3d6a85519e73c08acb4a | flake8 --diff -
+```
+
+**Please do not fix flake8 issues found in parts of the repository other than the bit that you are working on.** Not only is it very boring for you, but it is harder for maintainers to
+review your changes because so many of them are irrelevant to the hook you are adding or changing.
+
+
+### Add a news entry
+
 Please read [news/README.txt](https://github.com/pyinstaller/pyinstaller-hooks-contrib/blob/master/news/README.txt) before submitting you pull request.
+This will require you to know the pull request number before you make the pull request.
+You can usually guess it by adding 1 to the number of [the latest issue or pull request](https://github.com/pyinstaller/pyinstaller-hooks-contrib/issues?q=sort%3Acreated-desc).
+Alternatively, [submit the pull request](#submit-the-pull-request) as a draft,
+then add, commit and push the news item after you know your pull request number.
+
+
+### Summary
+
+A brief checklist for before submitting your pull request:
+
+* [ ] All new Python files have [the appropriate copyright header](#add-the-copyright-header).
+* [ ] You have written a [news entry](#add-a-news-entry).
+* [ ] Your changes [satisfy the linter](#run-linter) (run `git diff -U0 master | flake8 --diff -`).
+
+
+### Submit the pull request
+
+Once you've done all the above, go ahead and create a pull request.
+If you're stuck doing any of the above steps, create a draft pull request and explain what's wrong - we'll sort you out...
+Feel free to copy/paste commit messages into the Github pull request title and description.
+If you have run CI/CD, please include a link to it in your description so that we can see that it works.
+If you've never done a pull request before, note that you can edit it simply by running `git push` again.
```

Comment:
Not to rehash the whole discussion you already had with @rokm, but I think you should at least mention rebasing and force pushing. I had never used either of those things before working on this project and the majority of my time has been spent trying to clean my commits up, so I suspect others would also appreciate some advice or at least training resources.